spec/datadog/appsec/waf/converter_spec.rb

```diff
@@ -303,6 +303,12 @@
             expect(k).to eq "fooooooooo"
             expect(obj).to be_truncated
           end
+
+          it "converts hash with a nil key" do
+            obj = described_class.ruby_to_object({nil => :foo}, max_string_length: max_string_length)
```

Comment:
As it belongs to the truncation, could we add check that it will not be marked as truncated?

Additional question, what if the key is not a string? What if it's an object or number? Or how the nil sneak into the conversion?

Reply:
good point, the key could be basically anything - and we are converting it to a string later. I think I will move the conversion to happen earlier.

Reply:
I also noticed that symbols were incorrectly marked as truncated before: [`2e0c728` (#96)](https://github.com/DataDog/libddwaf-rb/pull/96/commits/2e0c72889a244238615cd745e418e15871815734#diff-b08d628c531f0660b9c26165feef32587e88dbdae2fd4f2409508a1353a07a1fR93-R98)